src/backend/tools/web_scrape.py

```diff
@@ -28,14 +30,34 @@ async def call(self, parameters: dict, **kwargs: Any) -> List[Dict[str, Any]]:
                 )
             ]
 
-        soup = BeautifulSoup(response.text, "html.parser")
-        text = soup.get_text(separator="\n")
-
-        return [
-            (
-                {
-                    "text": text,
-                    "url": url,
-                }
-            )
-        ]
+        content_type = response.headers.get("content-type")
+        if "application/pdf" in content_type:
```

Comment:
You can reuse the read_pdf function in the file service here

Reply:
Done